Galaxus.Functional/Async/AsyncResultExtensions.cs

```diff
@@ -0,0 +1,267 @@
+
+    /// <inheritdoc cref="Result{TOk,TErr}.OrElse{TContinuationErr}" />
+    public static async Task<Result<TOk, TContinuationErr>> OrElseAsync<TOk, TContinuationErr>(
+        this Task<Result<TOk, TContinuationErr>> self,
+        Func<TContinuationErr, Task<Result<TOk, TContinuationErr>>> continuation)
+    {
+        return await (await self).OrElseAsync(continuation);
+    }
+
+    /// <inheritdoc cref="Result{TOk,TErr}.Match{TResult}" />
+    public static async Task<TResult> MatchAsync<TOk, TErr, TResult>(
+        this Result<TOk, TErr> self,
+        Func<TOk, Task<TResult>> onOk,
+        Func<TErr, Task<TResult>> onErr)
+    {
+        return await self.Match(onOk, onErr);
+    }
+
+    /// <inheritdoc cref="Result{TOk,TErr}.Match{TResult}" />
+    public static async Task<TResult> MatchAsync<TOk, TErr, TResult>(
+        this Result<TOk, TErr> self,
+        Func<TOk, TResult> onOk,
+        Func<TErr, Task<TResult>> onErr)
+    {
+        return await self.Match(ok => Task.FromResult(onOk(ok)), onErr);
+    }
+
+    /// <inheritdoc cref="Result{TOk,TErr}.Match{TResult}" />
+    public static async Task<TResult> MatchAsync<TOk, TErr, TResult>(
+        this Result<TOk, TErr> self,
+        Func<TOk, Task<TResult>> onOk,
+        Func<TErr, TResult> onErr)
+    {
+        return await self.Match(onOk, err => Task.FromResult(onErr(err)));
+    }
+
+    /// <inheritdoc cref="Result{TOk,TErr}.Match{TResult}" />
+    public static async Task<TResult> MatchAsync<TOk, TErr, TResult>(
+        this Task<Result<TOk, TErr>> self,
+        Func<TOk, Task<TResult>> onOk,
+        Func<TErr, Task<TResult>> onErr)
+    {
+        return await (await self).Match(onOk, onErr);
+    }
+
+    /// <inheritdoc cref="Result{TOk,TErr}.Match{TResult}" />
+    public static async Task<TResult> MatchAsync<TOk, TErr, TResult>(
+        this Task<Result<TOk, TErr>> self,
+        Func<TOk, TResult> onOk,
+        Func<TErr, Task<TResult>> onErr)
+    {
+        return await (await self).Match(ok => Task.FromResult(onOk(ok)), onErr);
+    }
+
+    /// <inheritdoc cref="Result{TOk,TErr}.Match{TResult}" />
+    public static async Task<TResult> MatchAsync<TOk, TErr, TResult>(
+        this Task<Result<TOk, TErr>> self,
+        Func<TOk, Task<TResult>> onOk,
+        Func<TErr, TResult> onErr)
+    {
+        return await (await self).Match(onOk, err => Task.FromResult(onErr(err)));
+    }
+
+    /// <inheritdoc cref="Result{TOk,TErr}.Match{TResult}" />
+    public static async Task<TResult> MatchAsync<TOk, TErr, TResult>(
+        this Task<Result<TOk, TErr>> self,
+        Func<TOk, TResult> onOk,
+        Func<TErr, TResult> onErr)
+    {
+        return (await self).Match(onOk, onErr);
+    }
+
+    /// <summary>
+    ///     Alternative for <see cref="Result{TOk,TErr}.Match" />, producing another result.
+    /// </summary>
+    /// <inheritdoc cref="Result{TOk,TErr}.Match" />
+    [Obsolete("Use 'MatchAsync' instead and create the Result yourself")]
```

Comment:
I've included both `MatchResultAsync`-methods, but I'm unsure whether to actually continue them. In my book, it's just syntactical sugar for a rare case - I'd prefer creating the results explicitly.

What do you think, dear reviewer?